src/matching/Generic_vs_generic.ml

```diff
@@ -2800,6 +2800,20 @@ and m_list__m_type_parameter a b =
 
 and m_definition_kind a b =
   match (a, b) with
+  (* We maintain an equivalence between a FieldDefColon of an ellipsis
+     with any other definition. This is because FieldDefColon appears
+     when describing the RHS of a record, so this describes a record
+     like
+     { x = ... }
+     The FieldDefColon is only if it's not some other kind of
+     `definition_kind` structure, however, like a `FuncDef`, so we
+     would like to be able to match those with the ellipsis anyways.
```

Comment:
This one sentence is a bit confusing in wording, could use some cleaning up.

Great comment overall!